be/src/runtime/string_value.h

```diff
@@ -22,9 +22,53 @@
 
 #include "udf/udf.h"
 #include "util/hash_util.hpp"
+#include "util/cpu_info.h"
+#include "vec/common/string_ref.h"
+#ifdef __SSE4_2__
+#include "util/sse_util.hpp"
+#endif
 
 namespace doris {
 
+// Compare two strings using sse4.2 intrinsics if they are available. This code assumes
+// that the trivial cases are already handled (i.e. one string is empty).
+// Returns:
+//   < 0 if s1 < s2
+//   0 if s1 == s2
+//   > 0 if s1 > s2
+// The SSE code path is just under 2x faster than the non-sse code path.
+//   - s1/n1: ptr/len for the first string
+//   - s2/n2: ptr/len for the second string
+//   - len: min(n1, n2) - this can be more cheaply passed in by the caller
+static inline int string_compare(const char* s1, int64_t n1, const char* s2, int64_t n2,
```

Comment:
Why move it here?

Reply:
Because of the error in the link stage: The impl of `compare(const StringValue& other)` could not be found, but I don't know the reason, I will just move it to the string_value.h